### PR TITLE
react-components: Remove 'connectActive' method

### PR DESCRIPTION
**Pull request by bisgardo**
## Purpose

Remove the responsibility of initiating connections from `WithWalletConnector`. This simplifies this component and takes away any confusion as to why there's more than one way to establish a connection (and what exactly each method does and doesn't).

The field `activeConnection` will be moved from from this component into a hook in the followup PR #20. This will leave `WithWalletConnector` as a much simpler component with a single clear responsibility.

## Changes

Instead of using the `connectActive` method, one should just call `connect` directly on the connector which will then notify `WithWalletConnector` via its role as the connector's delegate. The application then calls `setActiveConnection` to set it as active if desired.

This means that `activeConnectorError` no longer holds errors for connections (only the connector's error as the name also implies) and that the `isConnecting` indicator no longer is maintained: It's now the application's job to track this state.

### CODE REVIEW

File: packages/react-components/src/WithWalletConnector.ts [hunk 1]
```diff
@@ -334,57 +291,20 @@ export class WithWalletConnector extends Component<Props, State> implements Wall
...
-        console.debug("WithWalletConnector: calling 'onDisconnect'", { connection, state: this.state });
+        console.debug("WithWalletConnector: calling 'onDisconnected'", { connection, state: this.state });
         this.setState((state) => ({
             ...state,
             activeConnection: connection === state.activeConnection ? undefined : state.activeConnection,
             connectedAccounts: updateMapEntry(state.connectedAccounts, connection, undefined),
         }));
     };
 
-    /**
```
Comment (bisgardo) [inline packages/react-components/src/WithWalletConnector.ts:348]: Add changelog entry.

**Comment by shjortConcordium** [comment]
I'm not sure I understand the purpose/reason for this change?

**Comment by bisgardo** [comment]
> I'm not sure I understand the purpose/reason for this change? thinking

Are you asking me whether you understand the reason? :stuck_out_tongue: On a serious note, I updated the stated purpose; is it clearer now?

**Review by shjortConcordium — APPROVED** [review]
Please update the README for react-components to describe how to connect, and remove the part where you use connectActive.

**Pull request merged**